gematria/datasets/exegesis_benchmark.cc

```diff
@@ -303,22 +324,35 @@ int main(int Argc, char *Argv[]) {
           errc::invalid_argument,
           "Malformed basic block: is not a JSON object"));
 
-    BenchmarkCode BenchCode = ExitOnErr(parseJSONBlock(
-        *AnnotatedBlockObject, *MachinePrinter, *MachineDisassembler, State));
+    Expected<BenchmarkCode> BenchCodeOrErr =
+        parseJSONBlock(*AnnotatedBlockObject, *MachinePrinter,
+                       *MachineDisassembler, State, BlockIndex);
+    if (!BenchCodeOrErr)
+      ExitOnErr(
+          createFileError(AnnotatedBlocksJson, BenchCodeOrErr.takeError()));
+
+    const BenchmarkCode &BenchCode = *BenchCodeOrErr;
 
     std::unique_ptr<const SnippetRepetitor> SnipRepetitor =
         SnippetRepetitor::Create(Benchmark::RepetitionModeE::MiddleHalfLoop,
                                  State, BenchCode.Key.LoopRegister);
 
-    unsigned Throughput100 =
-        ExitOnErr(benchmarkBasicBlock(BenchCode, *Runner, State));
+    Expected<double> ThroughputOrErr =
+        benchmarkBasicBlock(BenchCode, *Runner, State);
+
+    if (!ThroughputOrErr)
+      ExitOnErr(
+          createFileError(AnnotatedBlocksJson, ThroughputOrErr.takeError()));
```

Comment:
Is there some LLVM function that would do this generically, i.e. would create the file error when there _was_ an error, and reutrn `success()` otherwise?

If there's none, it feels like there should be one :)

Reply:
There's one in `llvm-exegesis`. It's pretty generic. I don't think there would be too much opposition to moving it to `Error.h`. That would requite a bit of work though. For now I've added `ExitOnFileErr` locally.